### PR TITLE
snapstate, ifacestate: add snapstate.CheckChangeConflict() to ifacestate.{Connect,Disconnect}

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: overlord/snapstate/snapstate.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ var Configure = func(st *state.State, snapName string, patch map[string]interfac
 	panic("internal error: snapstate.Configure is unset")
 }
 
-func checkChangeConflict(st *state.State, snapName string, snapst *SnapState) error {
+func CheckChangeConflict(st *state.State, snapName string, snapst *SnapState) error {
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: this merits a doc string now I think

File: overlord/ifacestate/ifacemgr.go [hunk 2]
```diff
@@ -118,17 +126,24 @@ func Connect(s *state.State, plugSnap, plugName, slotSnap, slotName string) (*st
...
+	confirmPlugConnection := hookstate.HookTask(st, summary, confirmPlugHookSetup, nil)
 	confirmPlugConnection.WaitFor(confirmSlotConnection)
 
 	return state.NewTaskSet(preparePlugConnection, prepareSlotConnection, connectInterface, confirmSlotConnection, confirmPlugConnection), nil
 }
 
 // Disconnect returns a set of tasks for  disconnecting an interface.
-func Disconnect(s *state.State, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
+func Disconnect(st *state.State, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
+	if err := snapstate.CheckChangeConflict(st, plugName, nil); err != nil {
```
Comment (pedronis) [inline overlord/ifacestate/ifacemgr.go]: this should be plugSnap, no? also no test?

File: overlord/ifacestate/ifacemgr.go [hunk 3]
```diff
@@ -118,17 +126,24 @@ func Connect(s *state.State, plugSnap, plugName, slotSnap, slotName string) (*st
...
+		return nil, err
+	}
+	if err := snapstate.CheckChangeConflict(st, slotName, nil); err != nil {
```
Comment (pedronis) [inline overlord/ifacestate/ifacemgr.go]: same

File: overlord/ifacestate/ifacemgr.go (hunk 2 above)
Comment (mvo5) [inline overlord/ifacestate/ifacemgr.go]: Sorry!

File: overlord/snapstate/snapstate.go [hunk 4]
```diff
@@ -214,7 +214,11 @@ var Configure = func(st *state.State, snapName string, patch map[string]interfac
 	panic("internal error: snapstate.Configure is unset")
 }
 
-func checkChangeConflict(st *state.State, snapName string, snapst *SnapState) error {
+// CheckChangeConflict ensures that for the given snapName no other changes
+// that alters the snap (like remove, install, refresh) is in progress. It
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: are in progress

File: overlord/snapstate/snapstate.go [hunk 5]
```diff
@@ -214,7 +214,11 @@ var Configure = func(st *state.State, snapName string, patch map[string]interfac
...
+// also ensures that snapst does not got modified. If a conflict happens
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: was not, or did not get

s/happens/is detected/ ?

Comment (pedronis) [inline overlord/snapstate/snapstate.go]: snapst (if not nil)

**Pull request merged**